### ISSUE TITLE
Doesn't work — Sketch 52

### ISSUE THREAD

**Issue by raulfunkie** [abynim/sketchy-pies#10]
The plugin seems to have stopped working with the recent update of Sketch 52.

It only shows the error below: 
<img width="532" alt="screen shot 2018-10-02 at 11 09 27 pm" src="https://user-images.githubusercontent.com/15929687/46388085-3e753f00-c698-11e8-8341-47be720dd3fc.png">

Is the plugin going to be updated or it's dead for good? (It would be such a shame!)

**Comment by ivomynttinen** [comment]
Same issue.

**Comment by elleck** [comment]
> The plugin seems to have stopped working with the recent update of Sketch 52.
> 
> It only shows the error below:
> <img alt="screen shot 2018-10-02 at 11 09 27 pm" width="532" src="https://user-images.githubusercontent.com/15929687/46388085-3e753f00-c698-11e8-8341-47be720dd3fc.png">
> 
> Is the plugin going to be updated or it's dead for good? (It would be such a shame!)

You already have the fix for it 😄 , enjoy your day!!!

**Comment by abynim** [comment]
Sorry folks, a lot has changed in Sketch 52. The issue reported here is an easy fix, but there are other issues causing pies not being generated as expected. I'll look into it, but can't promise an update this week.

**Comment by elleck** [comment]
Here is my fix https://github.com/elleck/sketchy-pies.git for Sketch 52, also made a small adjustment for the grouping

**Comment by abynim** [comment]
@elleck that does fix the errors, but it's not accurately drawing pies. For example, creating a pie with two colors should create two half circles, but it generates two full circles instead. The logic for drawing dash and gap seems to have changed in 52.

**Comment by elleck** [comment]
@abynim it working so far for @raulfunkie, but it is your plugin if you say it does not fix the error no problem, will wait for your fix

**Comment by abynim** [comment]
@elleck Thanks buddy. I appreciate the effort 👍 
My ocd won't let me release it without getting the details right! Hoping to get it fixed over the weekend.

**Comment by raulfunkie** [comment]
> @abynim it working so far for @raulfunkie, but it is your plugin if you say it does not fix the error no problem, will wait for your fix

Yes! Elleck's fix is working really well for me. Thanks so much!

**Comment by abynim** [comment]
@raulfunkie wow, that's strange! Can you confirm that making a pie with 2 colors gives you two half circles?
Here's a screen-cap of the bug I'm seeing: 
https://www.dropbox.com/s/d1hdxmy9l3q53lv/sketchypies-bug.mov?dl=0

**Comment by raulfunkie** [comment]
> @raulfunkie wow, that's strange! Can you confirm that making a pie with 2 colors gives you two half circles?
> Here's a screen-cap of the bug I'm seeing:
> https://www.dropbox.com/s/d1hdxmy9l3q53lv/sketchypies-bug.mov?dl=0

Yep. I actually never tried making the pie-chart using the first option, but just tried it and it does give me two half circles. The percentage version works fine too (50%/50% values).

**Comment by abynim** [comment]
Crazy stuff! Not sure what I'm missing. I'm using Elleck's build too, but no luck. You're testing with Sketch 52, right?

**Comment by raulfunkie** [comment]
> Crazy stuff! Not sure what I'm missing. I'm using Elleck's build too, but no luck. You're testing with Sketch 52, right?

Yes. Sketch 52 and as of 5mins ago 52.1 too.

https://www.dropbox.com/s/44h9lq1l7dsiyn5/sketchy-pie.mov?dl=0

**Comment by abynim** [comment]
Ok, I've asked the Sketch team about this. If they confirm that there is no change in logic then I'll merge in Elleck's code and just assume my Mac is going crazy. @elleck please send in a PR when you get a chance. Thanks guys 👍

**Comment by abynim** [comment]
Nevermind the PR @elleck, I've copied in your changes already 👍

**Comment by joaoplanche** [comment]
hy @abynim are there any new version that fixes this problem on 52?
Thank you very much for the amazing plugin, it's been saving a lot of ours!

**Comment by raulfunkie** [comment]
> hy @abynim are there any new version that fixes this problem on 52?
> Thank you very much for the amazing plugin, it's been saving a lot of ours!

hey @joaoplanche, you can try this version of the plugin. 
https://github.com/elleck/sketchy-pies.git

**Comment by b3hrooz** [comment]
Is this still working for people? I tried elleck's version also, but neither the original nor eleck's seem to be working for me on sketch 52.2

**Comment by TheBreznsoiza** [comment]
> > hy @abynim are there any new version that fixes this problem on 52?
> > Thank you very much for the amazing plugin, it's been saving a lot of ours!
> 
> hey @joaoplanche, you can try this version of the plugin.
> https://github.com/elleck/sketchy-pies.git

doesn’t work for me with v52.4 still the popup

> This only works when you have a Circle Shape layer selected. Width and Height must be equal.

**Comment by joanielutz** [comment]
elleck version isn't working for me either. Working on v52.3

**Comment by yannickdepauw** [comment]
Not working on version 52.6 - keep us posted with an update!

**Comment by kemurphy007** [comment]
I'm having the same issue on 52.6. Looking forward to an update, it's such a useful resource!

**Comment by ux-engineer** [comment]
Same issue with 53.

**Comment by DopDrop** [comment]
died

**Comment by iAmNaz** [comment]
dead

**Comment by abynim** [comment]
Sorry folks, I haven't been able to find a reliable solution to this issue. I would recommend using the Chart plugin instead https://github.com/pavelkuligin/chart

**Comment by rmarti55** [comment]
Still dead, great idea but bummer it didn't workout. Can you please note that on your code page/homepage? I downloaded the plugin before figuring out all these same issues everyone was having.
